metadata-ingestion/src/datahub/ingestion/source/ge_data_profiler.py

```diff
@@ -419,7 +419,14 @@ def _get_dataset_rows(self, dataset_profile: DatasetProfileClass) -> None:
                 self.dataset.engine.execute(get_estimate_script).scalar()
             )
         else:
-            dataset_profile.rowCount = self.dataset.get_row_count()
+            if (self.config.limit or self.config.offset) and not self.custom_sql:
+                # We don't want limit and offset to get applied to the row count
+                # This is kinda hacky way to do it, but every other way would require major refactoring
+                dataset_profile.rowCount = self.dataset.get_row_count(
+                    self.dataset_name.split(".")[-1]
+                )
+            else:
+                dataset_profile.rowCount = self.dataset.get_row_count()
```

Comment:
Can you add these two cases to the existing if / else block? Also, how does this work...? Can you elaborate in the comment because I don't get it lol